src/stability_sdk/client.py

```diff
@@ -171,6 +173,10 @@ def generate(
         guidance_strength: Optional[float] = None,
         guidance_prompt: Union[str, generation.Prompt] = None,
         guidance_models: List[str] = None,
+        upscale: bool = False,
+        upscale_kwargs: Optional[Dict[str, Any]] = None,
+        enhance: bool = False,
+        enhance_kwargs: Optional[Dict[str, Any]] = None,
```

Comment:
>a) as it is with bool/kwargs pairs

The logic behind this design is that the vast majority of devs will use the default settings. They will only care if upscale and enhance are true/false.

Current simplest upscale/enhance:
```
client.generate(..., 
    upscale=True,
    enhance=True)
```
or more options:
```
client.generate(..., 
    upscale=True,
    enhance=True,
    upscale_kwargs={'weight': 1.0} ,
    enhance_kwargs={'height': 1600})
```
# 

> b) pass list of post-processing requests to chain after 

Passing a list of request objects would look like:
```
upscale_rq = client._make_upscale_request()
enhance_rq = client._make_enhance_request()
postprocess_requests = [upscale_rq, enhance_rq]
answer = client.generate(..., 
    postprocess_requests=postprocess_requests)
```

#

>c) option on generate to return request so it can be separate chained (this is what api.py Context.generate does)

```
generate_rq = client.generate(..., 
    return_request=True)

upscale_rq = client.upscale(...,
    return_request=True)
enhance_rq = client.enhance(...,
    return_request=True)
requests_list = [generate_rq, upscale_rq, enhance_rq]
chain_rq = client._make_linear_chain(requests_list)
answer = client.run_request(chain_rq)
```
#
>d) upscale and enhance arguments as Union[bool, Dict[str, Any]] so True or a dict could be accepted

This is another simple intuitive option ( *** my favorite *** )

```
client.generate(..., 
    upscale=True,
    enhance=True)
```
or more options:
```
client.generate(..., 
    upscale={'weight': 1.0} ,
    enhance={'height': 1600})
```
#

>e) eliminate bools and pass *_kwargs empty dicts to enable them with defaults

Not quite as intuitive as method d
```
client.generate(..., 
    upscale={}, # this means we upscale
    enhance={}) # this means we enhance
```
#

>f) generate stays as is and enhance or other method added for performing chained generate+upscale/enhance request

```
client.generate_and_postprocess(..., 
    )
```